tests/unit/test_proto_events.py

```diff
@@ -80,6 +80,7 @@ def test_exception_events():
 
 
 def test_node_info_events():
+    meta_dict = {"string-key1": ["value1", 2], "string-key2": {"nested-dict-key": "value2"}, 1: "value-from-non-string-key"}
```

Comment:
Hi @gshank, thanks for the feedback! On your suggestion, I added a unit test to see if protobuf could serialize and de-serialize a more complicated meta object, including non-string keys and nested dictionaries.

The test passes, so if I understand this test correctly, I think it is correctly handling these cases. Does this satisfy your concerns, or should I look into protobuf structs?

Thanks!

Reply:
I misspoke before. The problem is with non-string values in the meta dictionary, not non-string keys. Also tests are failing.

Reply:
Hi @gshank, I fixed the broken test and also added tests for various kinds of non-string values and non-string keys in the meta dictionary: https://github.com/dbt-labs/dbt-core/pull/6493/files#diff-f9fe1fc7c69b0361dab53616751b87850948b5efd845478da472d2a68eec08afR83